src/CamelKJSONUtils.ts

```diff
@@ -126,3 +126,30 @@ export function shareMessage(outputChannel: vscode.OutputChannel, msg:string) {
 		console.log('[' + msg + ']');
 	}
 }
+
+export function toKebabCase (str : string) {
+	return str.replace(/([A-Z])([A-Z])/g, '$1-$2')
+		.replace(/([a-z])([A-Z])/g, '$1-$2')
+		.replace(/[\s_]+/g, '-')            
+		.toLowerCase() ;
+}
+
+// if a camel route file is selected in the explorer, it is opened in an editor.
+// if the user right-clicks to start a new integration, the same thing occurs. 
```

Comment:
if I right-click, the file is not selected for me so it will use the one that is already opened.
For instance, If I close all editors and view except the explorer. I right-click on a camel-k Groovy file and choose "Start Camel-K Integration", there is an undefined activeTextEditor:
![image](https://user-images.githubusercontent.com/1105127/61632723-d7a82d00-ac8d-11e9-904f-bf48299ac26f.png)

Reply:
there should be a callback to provide the file when the command is called, see https://stackoverflow.com/questions/51961457/how-to-get-file-name-or-path-in-vscode-extension-when-user-right-click-on-file-i (but cannot find it clear in official documentation https://code.visualstudio.com/api/references/vscode-api#commands )

Reply:
I'll investigate that. It was NOT very well documented how to get that

Reply:
Fixed it in this PR

Reply:
I think the function getSelectedEditorFile can be completely removed now

Reply:
(and it is a lot cleaner with the callback :-) )